Modules/SaltStackConfig/SaltStackConfig.psm1

```diff
@@ -289,3 +291,346 @@ Function Get-SscActivity {
+#>
+
+  (Get-SscData license get_current_license).desc
+}
+
+Function Get-SscvRALicense {
+<#
+  .NOTES
+  ===========================================================================
+    Created by:	Brian Wuchner
+    Date:		February 12, 2022
+    Blog:		www.enterpriseadmins.org
+    Twitter:		@bwuch
+  ===========================================================================
+  .SYNOPSIS
+    This wrapper function will return vRealize Automation license information for SaltStack Config.
+  .DESCRIPTION
+    This wrapper function will call Get-SscData license.get_vra_license and return the serial and edition property.
+  .EXAMPLE
+    PS C:\> Get-SscvRALicense
+#>
+
+  Get-SscData license get_vra_license
+}
+
+Function Get-SscMinionKeyState {
+  <#
+    .NOTES
+    ===========================================================================
+      Created by:	Brian Wuchner
+      Date:		February 12, 2022
+      Blog:		www.enterpriseadmins.org
+      Twitter:		@bwuch
+    ===========================================================================
+    .SYNOPSIS
+      This wrapper function will return minion key state information for SaltStack Config.
+    .DESCRIPTION
+      This wrapper function will call Get-SscData minions.get_minion_key_state and return the minions key states.  
+      Optionally a key state can be provided and the results will be filtered to only return the requested state.
+    .EXAMPLE
+      PS C:\> Get-SscMinionKeyState
+    .EXAMPLE
+      PS C:\> Get-SscMinionKeyState -key_state pending
+  #>
+  param(
+    [ValidateSet('accepted','rejected','pending','denied')][string]$key_state
+  )
+  
+  $kwarg = @{}
+  if ($key_state) { $kwarg.add('key_state',$key_state) }
+  
+  (Get-SscData minions get_minion_key_state $kwarg).results
+}
+
+
+Function Set-SscMinionKeyState {
+  <#
+    .NOTES
+    ===========================================================================
+      Created by:	Brian Wuchner
+      Date:		February 12, 2022
+      Blog:		www.enterpriseadmins.org
+      Twitter:		@bwuch
+    ===========================================================================
+    .SYNOPSIS
+      This wrapper function will set minion key state information for SaltStack Config.
+    .DESCRIPTION
+      This wrapper function will call Get-SscData minions.set_minion_key_state and update the states for specific minions.  
+    .EXAMPLE
+      PS C:\> Get-SscMinionKeyState |?{$_.name -eq 'server2022a'} | Set-SscMinionKeyState -state accept
+    .EXAMPLE
+      PS C:\> Set-SscMinionKeyState -master 'salt' -minion 'server2022a' -state reject -confirm:$false
+  #>
+  [cmdletbinding(SupportsShouldProcess)]
+  param(
+    [Parameter(Mandatory=$true, ValueFromPipeline=$true, ValueFromPipelineByPropertyName=$true)][string]$master,
+    [Parameter(Mandatory=$true, ValueFromPipeline=$true, ValueFromPipelineByPropertyName=$true)][string]$minion,
+    [Parameter(Mandatory=$true)][ValidateSet('accept','reject')][string]$state

```

Comment:
I agree that its a bit weird to mix tense between `Get-SscMinionKeyState` and `Set-SscMinionKeyState` input.  However, in the UI, when a minion key is selected, the action buttons available are accept key, reject key, delete key while the left navigation show values in the past tense.  Would it be appropriate to keep them as is in the module to line up with the UI?  I've included a screenshot below for reference.
![image](https://user-images.githubusercontent.com/1195173/154779838-ea464fbf-bb17-41fd-9c0e-85ec6d17e87f.png)

Reply:
I see what the situation is now. I think it would be best to rename cmdlets to Get-SscMinionKey, Set-SscMinionKey and Remove-SscMinionKey. Get-SscMinionKey should have -State filter that should accept 'Accepted', 'Pending', 'Rejected' or 'Denied'. Set-SscMinionKey should have two switch parameters -Accept and -Reject.

Reply:
I'll get this fixed in the next commit as well.  For `Set-SscMinionKey` I plan to use two ParameterSetNames, one for accept and another for reject, so that both switches cannot be specified concurrently.